core/src/main/java/org/mule/runtime/core/internal/config/FeatureFlaggingServiceBuilder.java

```diff
@@ -49,9 +51,17 @@ public FeatureFlaggingService build() {
 
     final String id = context.getConfiguration().getId();
     configurations.forEach((feature, p) -> {
-      boolean enabled = p.test(context);
-
-      LOGGER.debug("Setting feature {} = {} for artifact [{}]", feature, enabled, id);
+      boolean enabled;
+
+      String systemPropertyName = feature.getOverridingSystemPropertyName().orElse(null);
+      if (systemPropertyName != null && getProperty(systemPropertyName) != null) {
```

Comment:
why `systemPropertyName != null` here if you already have an optional in the previous line?